src/gafaelfawr/providers/oidc.py

```diff
@@ -199,6 +211,51 @@ async def logout(self, session: State) -> None:
         """
         pass
 
+    async def _get_ldap_uidNumber(self, uid: str) -> int:
+        """Retrieve the uid number from ldap
+
+        Parameters
+        ----------
+        uid : `str`
+            Username of the user.
+
+        Returns
+        -------
+        uidNumber : `int`
+            The numeric id of the user from LDAP.
+
+        Raises
+        ------
+        gafaelfawr.exceptions.LDAPException
+            The lookup using the uid_number_attr against the LDAP server was not
+            valid (attribute not in LDAP or resultant value not an integer).
+        """
+        assert self._ldap_config.uid_number_attr
+        attr = self._ldap_config.uid_number_attr
+        search = f"(&(Uid={uid}))"
+
+        self._logger.debug(
+            f"querying ldap {self._ldap_config.url} at {self._ldap_config.base_dn} for uid number with {search}"
```

Comment:
flake8 will insist that you break this into multiple strings with concatenation to keep the length down.

Reply:
fixed in bf08596ea67e12fc518bb883ae340b8c4b52a2bc